cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -452,8 +452,13 @@ WholeStageResultIteratorFirstStage::WholeStageResultIteratorFirstStage(
       auto partitionColumn = partitionColumns[idx];
       std::unordered_map<std::string, std::optional<std::string>> partitionKeys;
       constructPartitionColumns(partitionKeys, partitionColumn);
+#ifdef ENABLE_S3
+      auto path = UrlDecode(paths[idx]);
```

Comment:
Maybe, we can call a folly function to do the decoding.
See links:
https://github.com/facebook/folly/blob/775967aafc9c68af3d8f38619eac1cede43f0be7/folly/test/StringTest.cpp#L226C68-L226C68
https://github.com/facebook/folly/blob/775967aafc9c68af3d8f38619eac1cede43f0be7/folly/String-inl.h#L247

Reply:
@yma11 please check if this is necessary

Reply:
@yma11 could this patch also apply to gcs/abfs case?

Reply:
I think we also need this, we found this case caused failure for both localfs and abfs, when partition key contains '+'.

Reply:
We already do url decode in https://github.com/oap-project/gluten/blob/6d7d01fe0a433b82d3f216c992d244eb4686949a/backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/IteratorApiImpl.scala#L90
looks we will do it again, can we merge it to one place?